indexing-service/src/main/java/org/apache/druid/indexing/common/task/NoopTask.java

```diff
@@ -180,33 +155,18 @@ public int getPriority()
 
   public static NoopTask create()
   {
-    return new NoopTask(null, null, null, 0, 0, null, null, null);
+    return forDatasource(null);
   }
 
-  public static NoopTask withGroupId(String groupId)
+  public static NoopTask forDatasource(String datasource)
   {
-    return new NoopTask(null, groupId, null, 0, 0, null, null, null);
-  }
-
-  @VisibleForTesting
-  public static NoopTask create(String dataSource)
-  {
-    return new NoopTask(null, null, dataSource, 0, 0, null, null, null);
-  }
-
-  @VisibleForTesting
-  public static NoopTask create(int priority)
-  {
-    final Map<String, Object> context = new HashMap<>();
-    context.put(Tasks.PRIORITY_KEY, priority);
-    return new NoopTask(null, null, null, 0, 0, null, null, context);
+    return new NoopTask(null, null, datasource, 0, 0, null, null);
   }
 
-  @VisibleForTesting
-  public static NoopTask create(String id, int priority)
+  public static NoopTask withPriority(int priority)
```

Comment:
withPriority seems a bit misleading as one may think that the other fields' values would be retained

Reply:
We could rename it to `ofPriority`.